docs/developing-contracts/interactions-between-programs.md

```diff
@@ -1,197 +1,54 @@
 ---
-sidebar_label: Interaction Between Programs
+sidebar_label: Cross-program communication
 sidebar_position: 4
 ---
-# Cross-program communication
-
-This article explains how several programs (smart contracts) can communicate with each other by sending messages.
-
-Two simple programs are taken as an example:
-- `My Token` - this program will have an ability to mint tokens
-- `Wallet` - this program will store information about how many tokens the user has
-
-### `My Token` contract
-
-Let's write the contract `My Token` and the first step will be defining the contract struct that stores `name`, `symbol` and `balances` of accounts:
-
-```rust
-#[derive(Debug, Default, Encode, Decode, TypeInfo)]
-pub struct MyToken {
-    name: String,
-    symbol: String,
-    balances: BTreeMap<ActorId, u128>,
-}
-```
-
-Then we define `input` and `output` message types for contract initialization `init` and message handling `handle`.
 
-- The contract will be initialized with the following struct:
-
-    ```rust
-    #[derive(Debug, Encode, Decode, TypeInfo)]
-    pub struct InitConfig {
-        name: String,
-        symbol: String,
-    }
-    ```
-
-- The incoming messages will call this contract either for minting tokens or request an account balance:
-
-    ```rust
-    #[derive(Debug, Encode, Decode, TypeInfo)]
-    pub enum Action {
-        Mint(u128),
-        BalanceOf(ActorId),
-    }
-    ```
-
-- The outcoming messages will either report a successful mint or the user's balance:
-
-    ```rust
-    #[derive(Debug, Encode, Decode, TypeInfo)]
-    pub enum Event {
-        Minted {
-            to: ActorId,
-            amount: u128,
-        },
-        BalanceOf(u128),
-    }
-    ```
-
-It is necessary to define the message types in `metadata!` macro which is used to export correspondent functions from Rust:
-
-```rust
-gstd::metadata! {
-    title: "MyFungibleToken",
-    init:
-        input: InitConfig,
-    handle:
-        input: Action,
-        output: Event,
-}
-```
+# Cross-program communication
```

Comment:
Cross program invocation is not about async at all. It suits under telling about actor model. I'd provide separate section for async story in gear protocol

Reply:
Any suggestions?

Reply:
Done